utils/python/CIME/XML/machines.py

```diff
@@ -215,9 +214,7 @@ def is_valid_compiler(self,compiler):
         ...
         SystemExit: ERROR: COMPILERS value nag not supported for machine edison
         """
-        if self.get_field_from_list("COMPILERS", reqval=compiler) is not None:
-            return True
-        return False
+        return self.get_field_from_list("COMPILERS", reqval=compiler) is not None
```

Comment:
Yep, but I just noticed that the doctest above will fail now. I'll rerun doctest and pylint next week and clean this up.